### DISCUSSION TITLE
Grafik dengan 2 numerik

### DISCUSSION THREAD

**Discussion by joejnt** [teamalgoritma/community#53] [Q&A] [answered]
Bagaimana cara menyajikan dalam bentuk grafik untuk nominal transaksi tunai dan non tunai dari dataset ini
[BranchInformation.csv](https://github.com/teamalgoritma/community/files/8504327/BranchInformation.csv)

**Comment by tomytjandra** [comment, marked as answer]
Halo Pak @joejnt, misalkan di sini kita ingin memvisualisasikan jumlah transaksi cash maupun non-cash untuk masing-masing regional.

**Langkah:**
1. Membaca data. Fungsi `clean_names()` terdapat pada library `janitor` digunakan untuk menormalisasikan nama kolom agar mempermudah penulisan code berikutnya)

```
library(janitor)
branch <- read.csv("BranchInformation.csv", sep = ';') %>% 
  clean_names() # menormalkan nama kolom
str(branch)
```

2. Buatlah tabel agregasi yang menunjukkan jumlah `cash_trx_vol` dan jumlah `non_cash_trx_vol` untuk masing-masing `i_regional`

```
# cbind -> column bind
vol_regional <- aggregate(
  formula = cbind(cash_trx_vol, non_cash_trx_vol) ~ i_regional,
  data = branch,
  FUN = sum)

vol_regional
```
<img width="445" alt="image" src="https://user-images.githubusercontent.com/48079239/163793966-5fbc6486-3cc5-46e0-9d9c-163d310cf440.png">


3. Melakukan perubahan bentuk data (reshape) agar kolom `cash_trx_vol` dan `non_cash_trx_vol` dikumpulkkan menjadi 1 kolom dengan fungsi `pivot_longer()` dari library `tidyr`

```
library(tidyr)
data <- pivot_longer(vol_regional, cols = -i_regional, names_to = "type")
data
```
<img width="348" alt="image" src="https://user-images.githubusercontent.com/48079239/163793938-4f6fb3a8-2c3f-4ae6-a2f3-8825d03314ca.png">

5. Visualisasi dataframe `vol_regional` menjadi grouped bar chart dengan library `ggplot2` (kita akan mempelajarinya di course DV)

```
library(ggplot2)
ggplot(data, aes(x = i_regional, y = value, fill = type)) +
  geom_bar(stat = "identity", position = "dodge")
```

Penjelasan parameter:
- `aes()` untuk mendefinisikan peletakkan plot dengan sumbu `x` dari kolom `i_regional` dan sumbu `y` dari kolom `value`. Warna bar dibedakan berdasarkan kolom `type` (yaitu cash dan non_cash)
- `geom_bar()` digunakan untuk membuat diagram batangnya

![image](https://user-images.githubusercontent.com/48079239/163794017-53471f88-347a-4ba6-80cd-af408bf902a7.png)

**Reply by joejnt** [reply]
well noted, thank you
